domain/src/main/java/org/oppia/android/domain/question/QuestionAssessmentCalculation.kt

```diff
@@ -0,0 +1,118 @@
+package org.oppia.android.domain.question
+
+import org.oppia.android.app.model.FractionGrade
+import org.oppia.android.app.model.UserAssessmentPerformance
+import javax.inject.Inject
+import kotlin.math.max
+
+/**
+ * Private class that computes the state of the user's performance at the end of a practice session.
+ * This class can exist across multiple training session instances, but calling code is responsible
+ * for ensuring it is properly reset. This class is not thread-safe, so it is the calling code's
+ * responsibility to synchronize access to this class.
+ */
+internal class QuestionAssessmentCalculation constructor(
```

Comment:
```suggestion
internal class QuestionAssessmentCalculation private constructor(
```

Since this is only called by the factory class (and should be).

Reply:
Fixed